pytest/tests/loadtest/locust/common/base.py

```diff
@@ -145,12 +145,38 @@ class NearNodeProxy:
     Wrapper around a RPC node connection that tracks requests on locust.
     """
 
-    def __init__(self, host, request_event):
-        self.request_event = request_event
-        url, port = host.split(":")
+    def __init__(self, environment):
+        self.request_event = environment.events.request
+        url, port = environment.host.split(":")
         self.node = cluster.RpcNode(url, port)
         self.session = requests.Session()
 
+    def send_tx_retry(self, tx: Transaction, locust_name):
+        """
+        Send a transaction and retry until it succeeds
+        
+        This method retries no matter the kind of error, but it tries to be
+        smart about what to do depending on the error.
+        
+        Expected error: UnknownTransactionError means TX has not been executed yet.
+        Expected error: InvalidNonceError means we are using an outdated nonce.
+        Other errors: Probably bugs in the test setup (e.g. invalid signer).
+        """
+        while True:
+            try:
+                result = self.send_tx(tx, locust_name)
```

Comment:
No need for an auxiliary variable here, you can just do `return self.send_tx(tx, locust_name)`